packages/config/src/ios/Updates.ts

```diff
@@ -83,3 +87,129 @@ export function setVersionsConfig(config: ExpoConfig, expoPlist: ExpoPlist) {
+  const buildScriptPath = projectHasModule(
+    'expo-updates/scripts/create-manifest-ios.sh',
+    projectDir,
+    config
+  );
+
+  if (!buildScriptPath) {
+    throw new Error(
+      "Could not find the build script for iOS. This could happen in case of outdated 'node_modules'. Run 'npm install' to make sure that it's up-to-date."
+    );
+  }
+
+  return path.relative(path.join(projectDir, 'ios'), buildScriptPath);
+}
+
+interface ShellScriptBuildPhase {
+  isa: 'PBXShellScriptBuildPhase';
+  name: string;
+  shellScript: string;
+  [key: string]: any;
+}
+
+function getBundleReactNativePhase(project: xcode.XcodeProject): ShellScriptBuildPhase {
+  const scriptBuildPhase = project.hash.project.objects.PBXShellScriptBuildPhase as Record<
+    string,
+    ShellScriptBuildPhase
+  >;
+  const bundleReactNative = Object.values(scriptBuildPhase).find(
+    buildPhase => buildPhase.name === '"Bundle React Native code and images"'
+  );
+
+  if (!bundleReactNative) {
+    throw new Error(`Couldn't find a build phase "Bundle React Native code and images"`);
+  }
+
+  return bundleReactNative;
+}
+
+export function setBuildPhaseShellScript(
+  projectDir: string,
+  config: ExpoConfig,
+  project: xcode.XcodeProject
+): xcode.XcodeProject {
+  const bundleReactNative = getBundleReactNativePhase(project);
+  const buildPhaseShellScriptPath = getBuildPhaseShellScriptPath(projectDir, config);
+
+  if (!bundleReactNative.shellScript.includes(buildPhaseShellScriptPath)) {
+    bundleReactNative.shellScript = `${bundleReactNative.shellScript.replace(
+      /"$/,
+      ''
+    )}${buildPhaseShellScriptPath}\\n"`;
+  }
+  return project;
+}
+
+export function hasBuildPhaseShellScript(
+  projectDir: string,
+  config: ExpoConfig,
+  project: xcode.XcodeProject
+): boolean {
+  const bundleReactNative = getBundleReactNativePhase(project);
+  const buildPhaseShellScriptPath = getBuildPhaseShellScriptPath(projectDir, config);
+
+  return bundleReactNative.shellScript.includes(buildPhaseShellScriptPath);
+}
+
+export function isPlistConfigurationSet(expoPlist: ExpoPlist): boolean {
+  if (!expoPlist.EXUpdatesURL) {
+    return false;
+  }
+
+  if (!expoPlist.EXUpdatesSDKVersion && !expoPlist.EXUpdatesRuntimeVersion) {
+    return false;
+  }
+  return true;
+}
+
+export function isPlistConfigurationSynced(
+  config: ExpoConfig,
+  expoPlist: ExpoPlist,
+  username: string | null
+): boolean {
+  if (getUpdateUrl(config, username) !== expoPlist.EXUpdatesURL) {
+    return false;
+  }
+  if (getUpdatesEnabled(config) !== expoPlist.EXUpdatesEnabled) {
+    return false;
+  }
+  if (getUpdatesTimeout(config) !== expoPlist.EXUpdatesLaunchWaitMs) {
+    return false;
+  }
+  if (getUpdatesCheckOnLaunch(config) !== expoPlist.EXUpdatesCheckOnLaunch) {
+    return false;
+  }
+
+  if (!isPlistVersionConfigurationSynced(config, expoPlist)) {
+    return false;
+  }
+
+  return true;
```

Comment:
Please refactor this. It's hard to read.

Especially, this `if` doesn't make sense.
```
 if (!isPlistVersionConfigurationSynced(config, expoPlist)) {
    return false;
  }
```